mas_perception_libs/common/include/mas_perception_libs/bounding_box_2d.h

```diff
@@ -0,0 +1,105 @@
+#include <utility>
+
+/*!
+ * @copyright 2018 Bonn-Rhein-Sieg University
+ *
+ * @author Minh Nguyen
+ *
+ */
+#ifndef MAS_PERCEPTION_LIBS_BOUNDING_BOX_2D_H
+#define MAS_PERCEPTION_LIBS_BOUNDING_BOX_2D_H
+
+#include <string>
+#include <vector>
+#include <opencv/cv.h>
+
+namespace mas_perception_libs
+{
+
+struct BoundingBox2D
+{
+    /*!
+     * @brief struct for representing a rectangle region in RGB images
+     */
+    int mX;
+    int mY;
+    int mWidth;
+    int mHeight;
+    std::string mLabel;
+    cv::Scalar mColor;
+
+    BoundingBox2D(std::string pLabel, cv::Scalar pColor, int pX, int pY, int pWidth, int pHeight)
+            : mX(pX), mY(pY), mWidth(pWidth), mHeight(pHeight), mLabel(std::move(pLabel)), mColor(pColor)
+    { }
+
+    BoundingBox2D(std::string pLabel, int pX, int pY, int pWidth, int pHeight)
```

Comment:
I think passing arguments by const reference is best practice for non-primitive datatypes (i.e.  `const std::string &pLabel` and `const cv::Scalar &pColor`)

https://www.learncpp.com/cpp-tutorial/73-passing-arguments-by-reference/

Can you also look at all your other function arguments?

Reply:
updated all unmodified params to const reference with the last commit.